x-pack/test/plugin_functional/test_suites/resolver/index.ts

```diff
@@ -6,22 +6,101 @@
 
 import { FtrProviderContext } from '../../ftr_provider_context';
 
-export default function ({ getPageObjects, getService }: FtrProviderContext) {
+export default function ({
+  getPageObjects,
+  getService,
+  updateBaselines,
+}: FtrProviderContext & { updateBaselines: boolean }) {
   const pageObjects = getPageObjects(['common']);
   const testSubjects = getService('testSubjects');
+  const screenshot = getService('screenshots');
+  const find = getService('find');
+  const browser = getService('browser');
 
-  describe('Resolver embeddable test app', function () {
+  describe('Resolver test app', function () {
     this.tags('ciGroup7');
 
     beforeEach(async function () {
       await pageObjects.common.navigateToApp('resolverTest');
     });
 
-    it('renders a container div for the embeddable', async function () {
-      await testSubjects.existOrFail('resolverEmbeddableContainer');
-    });
-    it('renders resolver', async function () {
-      await testSubjects.existOrFail('resolverEmbeddable');
+    it('renders at least one node, one node-list, one edge line, and graph controls', async function () {
+      await testSubjects.existOrFail('resolver:node');
+      await testSubjects.existOrFail('resolver:node-list');
+      await testSubjects.existOrFail('resolver:graph:edgeline');
+      await testSubjects.existOrFail('resolver:graph-controls');
+
+      // stuff i'd like to test
+      // nodes: short name origin node with 13 pills, long name non-origin node, short name non-origin node
+      // node states: node is selected, node is unselected
+      // interaction states: node button is hovered, node button is focused, node button is hovered and focused, pill is focused, pill is hovered, pill is hovered and focused
+
+      // make the window big enough that all nodes are fully in view (for screenshots)
+      await browser.setWindowSize(1920, 1200);
+
+      // Because the lint rules will not allow files that include upper case characters, we specify explicit file name prefixes
+      const nodeDefinitions: Array<[nodeID: string, fileNamePrefix: string]> = [
+        ['origin', 'origin'],
+        ['firstChild', 'first_child'],
+        ['secondChild', 'second_child'],
+      ];
+
+      for (const [nodeID, fileNamePrefix] of nodeDefinitions) {
+        const element = await find.byCssSelector(`[data-test-resolver-node-id="${nodeID}"]`);
+        await screenshot.compareAgainstBaseline(`${fileNamePrefix}`, updateBaselines, element);
```

Comment:
This compares the screenshot to the baseline but I do not see where you are validating the results of the compare.  `compareAgainstBaseline()` returns a percentage difference between the two images.  Seems like you should have an `expect()` to determine what percentage difference is acceptable.

Reply:
@charlie-pichette Thank you very much for the catch! I was just so excited to see screenshots 👍